src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,152 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size = 0;
+
+    public MyLinkedList() {
+        this.head = null;
+        this.tail = null;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> oldTail = tail;
+        Node<T> newNode = new Node<>(oldTail, value, null);
+        tail = newNode;
+        if (oldTail == null) {
+            head = newNode;
+        } else {
+            oldTail.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index: " + index + ", Size: " + size);
+        }
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            Node<T> oldHead = head;
+            head = new Node<>(null, value, oldHead);
+            if (oldHead == null) {
+                tail = head;
+            }
+            oldHead.prev = head;
+            size++;
+        } else {
+            Node<T> actualNode = head;
+            for (int i = 0; i < index; i++) {
+                actualNode = actualNode.next;
+            }
+            Node<T> newNode = new Node<>(actualNode.prev, value, actualNode);
+            actualNode.prev.next = newNode;
+            actualNode.prev = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        list.forEach(element -> add(element));
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkElementIndex(index);
+        Node<T> actualNode = head;
+        for (int i = 0; i < index; i++) {
+            actualNode = actualNode.next;
+        }
```

Comment:
it's not good to iterate from beginning only, if we have 10000 element and index 9950 it's much faster to go from tail.
Please recheck same behaviour in other methods.
Probably extract node search logic to separate method